mockbroker.go

```diff
@@ -192,8 +215,10 @@ func (b *MockBroker) handleRequests(conn net.Conn, idx int, wg *sync.WaitGroup)
 		}
 
 		b.lock.Lock()
+		originalHandlerUsed := b.origHandler
```

Comment:
Not a huge fan of all this code:
- I'm not sure what it's for, the tests already failed if an expectation was not consumed?
- Is there a way to encapsulate the expectation-counting code inside the handler (perhaps via a closure) so that we don't have to track whether we're using the original or not, the handler just does the right thing?